stage1/init/common/mount.go

```diff
@@ -294,8 +296,29 @@ func AppAddMounts(p *stage1commontypes.Pod, ra *schema.RuntimeApp, enterCmd []st
 		os.Exit(254)
 	}
 
+	absRoot, err := filepath.Abs(p.Root)
+	if err != nil {
+		log.FatalE("could not determine pod's absolute path", err)
+	}
+
+	appRootfs := common.AppRootfsPath(absRoot, ra.Name)
+
 	for _, m := range mounts {
-		AppAddOneMount(p, ra, m.Volume.Source, m.Mount.Path, m.ReadOnly, enterCmd)
+		shPath := filepath.Join(common.SharedVolumesPath(p.Root), m.Volume.Name.String())
+		mntPath, err := EvaluateSymlinksInsideApp(appRootfs, m.Mount.Path)
```

Comment:
While that comment is technically correct, we still need the real mount path ourselves - so we can't get rid of it.